### ISSUE TITLE
Resulting PMML from Catboost conversion is very large

### ISSUE THREAD

**Issue by rurao** [jpmml/jpmml-catboost#7]
Hi,

We are trying to convert catboost models to PMML and see that the PMML model is an order of magnitude larger than the catboost model. Attaching a sample catboost model that is 38MB which when converted turns out to be a 823MB PMML model (reduces to 560MB when removing whitespaces).
[refit_STARSv3_55var_cbm_model.cbm.zip](https://github.com/user-attachments/files/16617701/refit_STARSv3_55var_cbm_model.cbm.zip)

Wondering if it just that the there are a lot of TreeModels and TreeModel representation in PMML is too verbose or are there any opportunities to reducing the size of the final PMML.

**Comment by vruusmann** [comment]
> We see that the PMML model is an order of magnitude larger than the catboost model.

You're comparing an optimized binary data format (CBM) against a human-oriented text file format (PMML). What did you expect to see?

> Attaching a sample catboost model that is 38MB which when converted turns out to be a 823MB PMML model

The sample CBM model converted to a 785 MB PMML model file on my computer. After compression (`zip -9 model.pmml`), its size shrinks to 33.5 MB.

Now, you have a 18 MB CBM+ZIP vs a 34 MB PMML+ZIP. Using this metric, PMML appears two times more verbose than CBM.

Instead of measuring static files, you should measure the memory consumption of the application after the model has been loaded from file to memory.

> or are there any opportunities to reducing the size of the final PMML.

It's possible to re-arrange `Node` elements from (CatBoost-style-) binary splits to PMML-style multi-way splits. This compaction is implemented for LightGBM and XGBoost converters, but not for CatBoost.

This re-arrangement shrings PMML file size anywhere between 30 - 50% (depending on how much features are reused).

**Comment by vruusmann** [comment]
BTW, the JPMML-Catboost project is effectively in "archived" state.

The CatBoost project provides its own PMML exporter. I have zero interest in competing with vendor-backed functionality.
